samza-core/src/main/java/org/apache/samza/processor/SamzaContainerController.java

```diff
@@ -106,7 +109,17 @@ public void startContainer(ContainerModel containerModel, Config config, int max
         Util.<String, MetricsReporter>javaMapAsScalaMap(metricsReporterMap),
         taskFactory);
     log.info("About to start container: " + containerModel.getContainerId());
-    containerFuture = executorService.submit(() -> container.run());
+    containerFuture = executorService.submit(() -> {
+        try {
+          container.run();
+        } catch (Throwable t) {
+          if (errorHandler != null) {
+            errorHandler.onProcessorError(t);
+          } else {
+            throw t;
```

Comment:
We can log error. I actually didn't know what to do in the else block, because even if you throw here, the user doesn't have the handle to the container's future. So, its pretty useless. I think logging and then, throwing is probably better. Let me add that.